Sources/NIOFileSystem/FileHandleProtocol.swift

```diff
@@ -202,6 +202,13 @@ public protocol ReadableFileHandleProtocol: FileHandleProtocol {
     ///   - chunkLength: The maximum length of the chunk to read as a ``ByteCount``.
     /// - Returns: A sequence of chunks read from the file.
     func readChunks(in range: Range<Int64>, chunkLength: ByteCount) -> FileChunks
+
+    /// Returns an asynchronous sequence of chunks read from the file starting from the current file pointer.
+    ///
+    /// - Parameters:
+    ///   - size: The maximum length of the chunk to read as a ``ByteCount``.
+    /// - Returns: A sequence of chunks read from the file.
+    func readChunksFromFilePointer(chunkLength size: ByteCount) -> FileChunks
```

Comment:
This isn't quite what I had in mind. Rather than adding new API I think we should use the type of the file to determine how to do the read inside `FileChunks`. Once we know the type of the file we can determine whether the range passed in is acceptable and then call the appropriate read function.

Reply:
This would make the checks that happen inside of `.readToEnd` redundant. Should we keep those or remove them?

Reply:
This also means that calling readToEnd will stat the file twice.

I’m trying out your recommended solution and it also causes problems because if we check the file type in the FileChunks initializer, then that means the function has to be async throws. But the readChunks function from `ReadableFileHandleProtocol` is neither async or throws so that would be an API change.